Formula/heimdal.rb

```diff
@@ -31,6 +30,8 @@ class Heimdal < Formula
 
   uses_from_macos "perl"
 
+  conflicts_with "krb5", because: "both install `gssapi.h`"
```

Comment:
As mentioned in comments, I think a `keg_only` on Linux would be better.

Not sure if there is ideal way to specify this
- For specific reason: macOS `keg_only` due to shadowed & Linux `keg_only` due to conflict with `krb5`. But I don't know if Linux-only `keg_only` works
- Or just general `keg_only` for all OS.